doc/modules/grid_search.rst

```diff
@@ -167,6 +171,358 @@ variable that is log-uniformly distributed between ``1e0`` and ``1e3``::
+
+Notice in the example above that the last iteration does not use the maximum
+amount of resources available: 1000 samples are available, yet only 640 are
+used, at most. By default, ``min_resources`` is set to a specific value such
+that the last iteration uses as many samples as possible. Please see
+:ref:`exhausting_the_resources` for details.
+
+.. _amount_of_resource_and_number_of_candidates:
+
+Amount of resource and number of candidates at each iteration
+-------------------------------------------------------------
+
+At any iteration `i`, each candidate is allocated a given amount of resources
+which we denote `resource_iter`. This quantity is controlled by the
+parameters ``ratio`` and ``min_resources`` as follows (`ratio` is strictly
+greater than 1)::
+
+    resource_iter = ratio**i * min_resources,
+
+where ``min_resources`` is the amount of resources used at the first
+iteration. ``ratio`` also defines the proportions of candidates that
+will be selected for the next iteration::
+
+    n_candidates_iter = n_candidates // (ratio ** i)
+
+or equivalently::
+
+    n_candidates_at_i+1 = n_candidates_at_i // ratio
+
+So in the first iteration, we use ``min_resources`` resources
+``n_candidates`` times. In the second iteration, we use ``min_resources *
+ratio`` resources ``n_candidates // ratio`` times. The third again
+multiplies the resources per candidate and divides the number of candidates.
+This process stops when the maximum amount of resource per candidate is
+reached, or when we have identified the best candidate. The best candidate
+is identified at the iteration that is evaluating `ratio` or less candidates
+(see just below for an explanation).
+
+Here is an example with ``min_resources=3`` and ``ratio=2``, starting with
+70 candidates:
+
++-----------------------+-----------------------+
+| ``resource_iter``     | ``n_candidates_at_i`` |
++=======================+=======================+
+| 3 (=min_resources)    | 70 (=n_candidates)    |
++-----------------------+-----------------------+
+| 3 * 2 = 6             | 70 // 2 = 35          |
++-----------------------+-----------------------+
+| 6 * 2 = 12            | 35 // 2 = 17          |
++-----------------------+-----------------------+
+| 12 * 2 = 24           | 17 // 2 = 8           |
++-----------------------+-----------------------+
+| 24 * 2 = 48           | 8 // 2 = 4            |
++-----------------------+-----------------------+
+| 48 * 2 = 96           | 4 // 2 = 2            |
++-----------------------+-----------------------+
+
+We can note that:
+
+- the process stops at the first iteration which evaluates `ratio=2`
+  candidates: the best candidate is the best out of these 2 candidates. It
+  is not necessary to run an additional iteration, since it would only
+  evaluate one candidate (namely the best one, which we have already
+  identified). For this reason, **in general, we want the last iteration to
+  run at most ``ratio`` candidates**.
```

Comment:
I'm not sure I agree with that. Why "at most"? We get the maximum speed-up in that case but if we want to spend more resources doing it differently is fine, right?

Reply:
The way I understand it is that if you evaluate more than `ratio` candidates at the last iteration, then you didn't perform SH: you performed SH up to the second to last iteration, and then the last iteration was just a regular search.


I understand SH as being a tournament where each iteration should have the same rules. If you evaluate more than ratio candidates at the last iteration, that iteration has different rules because it selects the top-1 candidate instead of the top-k.

I'll remove the bold font though ^^